modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/plugins/fields-sidebar/components/FieldsSidebarSettingsBody.es.js

```diff
@@ -12,113 +12,99 @@
  * details.
  */
 
-import React, {
-	useContext,
-	useEffect,
-	useLayoutEffect,
-	useRef,
-	useState,
-} from 'react';
+import {
+	EVENT_TYPES,
+	FormProvider,
+	FormRenderer,
+} from 'dynamic-data-mapping-form-renderer';
+import React, {useContext, useEffect, useMemo, useState} from 'react';
 
 import AppContext from '../../../AppContext.es';
 import {EDIT_CUSTOM_OBJECT_FIELD} from '../../../actions.es';
 import DataLayoutBuilderContext from '../../../data-layout-builder/DataLayoutBuilderContext.es';
-import renderSettingsForm, {
-	getEvents,
-	getFilteredSettingsContext,
-} from '../../../utils/renderSettingsForm.es';
+import {getFilteredSettingsContext} from '../../../utils/settingsForm.es';
+
+const spritemap = `${Liferay.ThemeDisplay.getPathThemeImages()}/lexicon/icons.svg`;
```

Comment:
I'm not sure if we have a ClayIcon Context above the component, it was something that existed before, so I just copied it. I will test to eliminate this.

Reply:
Yeah, this works, I thought we weren't using React 😅 taglib. I thought about not even sending the spritemap to FormRenderer, since it is in the same structure, so we don't need it but then I remembered that we can load a field in Metal yet 😞.